File: SAMSUI/src/app/Model/master/asset-group.ts
import { AssetCustFieldValue } from "./asset-cust-field-value";
import { BaseModel } from "../base/baseModel";

export class AssetGroupModel extends BaseModel {

    public assetGroupId ?: number;

    public assetGroupName ?: string;

    public assetGroupDesc ?: string;

    public assetTypeId ?: number;

    public assetTypeName ?: string;

    public assetCategoryId ?: number;

    public assetCategoryName ?: string;

    public subCategoryId ?: number;

    public subCategoryName ?: string;

    public orgId ?: number;

    public active ?: boolean;

    public createdBy ?:string;

    public createdDtDisp ?: string;

    public updatedBy ?: string;

    public updatedDt ?: Date;

    public createdDt ?: Date;

    public updatedDtDisp ?: string;

    public pageNumber ?: number;

    public recordsPerPage ?: number;

    public orgName ?: string;

    public frequency1 ?: string;

    public frequency2 ?: string; 

    public frequency3 ?: string; 

    public maintenanceStrategy ?: string;

    public criticalNature ?: string;

    public emScore ?: string;

    public childAsset ?: boolean;

    public assetCustomFieldValue ?:AssetCustFieldValue[];

    public assetGroupAttribute1 ?: string;
    public assetGroupAttribute2 ?: string;
    public assetGroupAttribute3 ?: string;
    public assetGroupAttribute4 ?: string;
    public assetGroupAttribute5 ?: string;

    public columnName ?: string;
    public direction ?: string;

    public isListEmpty ?:boolean;

    public specification ?: boolean;

    public deviceCode ?: string;

    public deviceConcept ?: string;

    public assetGroupCode ?:string;

}

File: sams-backend/src/main/java/com/sams/entity/AssetGroup.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_asset_group", schema = "asset")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AssetGroup {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "asset_group_id")
    private Long assetGroupId;

    @Column(name = "asset_group_name")
    private String assetGroupName;

    @Column(name = "asset_group_desc")
    private String assetGroupDesc;

    @Column(name = "asset_type_id")
    private Long assetTypeId;

    @Column(name = "asset_type_name")
    private String assetTypeName;

    @Column(name = "asset_category_id")
    private Long assetCategoryId;

    @Column(name = "asset_category_name")
    private String assetCategoryName;

    @Column(name = "sub_category_id")
    private Long subCategoryId;

    @Column(name = "sub_category_name")
    private String subCategoryName;

    @Column(name = "org_id")
    private Long orgId;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt_disp")
    private String createdDtDisp;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt")
    private LocalDateTime updatedDt;

    @Column(name = "created_dt")
    private LocalDateTime createdDt;

    @Column(name = "updated_dt_disp")
    private String updatedDtDisp;

    @Column(name = "org_name")
    private String orgName;

    @Column(name = "frequency1")
    private String frequency1;

    @Column(name = "frequency2")
    private String frequency2;

    @Column(name = "frequency3")
    private String frequency3;

    @Column(name = "maintenance_strategy")
    private String maintenanceStrategy;

    @Column(name = "critical_nature")
    private String criticalNature;

    @Column(name = "em_score")
    private String emScore;

    @Column(name = "child_asset")
    private Boolean childAsset;

    @Column(name = "asset_custom_field_value")
    private String assetCustomFieldValue;

    @Column(name = "asset_group_attribute1")
    private String assetGroupAttribute1;

    @Column(name = "asset_group_attribute2")
    private String assetGroupAttribute2;

    @Column(name = "asset_group_attribute3")
    private String assetGroupAttribute3;

    @Column(name = "asset_group_attribute4")
    private String assetGroupAttribute4;

    @Column(name = "asset_group_attribute5")
    private String assetGroupAttribute5;

    @Column(name = "column_name")
    private String columnName;

    @Column(name = "direction")
    private String direction;

    @Column(name = "specification")
    private Boolean specification;

    @Column(name = "device_code")
    private String deviceCode;

    @Column(name = "device_concept")
    private String deviceConcept;

    @Column(name = "asset_group_code")
    private String assetGroupCode;

    @PrePersist
    protected void onCreate() {
       // Auto-generated
    }

    @PreUpdate
    protected void onUpdate() {
       // Auto-generated
    }
}